deepchem/utils/pytorch_utils.py

```diff
@@ -14,3 +14,48 @@ def get_activation(fn: Union[Callable, str]):
     if isinstance(fn, str):
         return getattr(torch.nn.functional, fn)
     return fn
+
+
+def unsorted_segment_sum(data, segment_ids, num_segments):
+    """Computes the sum along segments of a tensor. Analogous to tf.unsorted_segment_sum.
+
+    Parameters
+    ----------
+    data: torch.Tensor
+        A tensor whose segments are to be summed.
+    segment_ids: torch.Tensor
+        The segment indices tensor.
+    num_segments: int
+        The number of segments.
+
+    Returns
+    -------
+    tensor: torch.Tensor
+
+    Examples
+    --------
+    >>> segment_ids = torch.Tensor([0, 1, 0]).to(torch.int64)
+    >>> data = torch.Tensor([[1, 2, 3, 4], [5, 6, 7, 8], [4, 3, 2, 1]])
+    >>> num_segments = 2
+    >>> result = unsorted_segment_sum(data=data,
+                                  segment_ids=segment_ids,
+                                  num_segments=num_segments)
+    >>> result
+    tensor([[5., 5., 5., 5.],
+        [5., 6., 7., 8.]])
+
+    """
+    # segment_ids.shape should be a prefix of data.shape
+    assert all([i in data.shape for i in segment_ids.shape])
```

Comment:
I don't this this works correctly; it checks if every index size in segment_ids.shape appears in data.shape, and not that segment_ids.shape is a prefix of data.shape

Reply:
Oh right, I have updated it now.